onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -578,7 +578,13 @@ static void RegisterExecutionProviders(InferenceSession* sess, const std::vector
       auto it = provider_options_map.find(type);
       if (it != provider_options_map.end()) {
         for (auto option : it->second) {
-          if (option.first == "has_trt_options") {
+          if (option.first == "device_id") {
+            if (!option.second.empty()) {
+              params.device_id = std::stoi(option.second);
+            } else {
+              ORT_THROW("[ERROR] [TensorRT] The value for the key 'device_id' should be a number i.e. '0'.\n");
+            }
+          } else if (option.first == "has_trt_options") {
```

Comment:
when I look at this api again, this looks odd.
it's not very user friendly that the user has to set an option called "has_trt_options" to tell us that they have set other options.
By virtue of provider_options_map containing items, that is telling us that the user has set some options, right?
can't we just check the size of provider_options_map to tell if the user has set options?

we also need to fix this on the c api side.
there shouldn't be a "has_trt_options" field in OrtTensorRTProviderOptions
In Tensorrt_provider::CreateExecutionProviderFactory(provider_options),
we can implictly set info.has_trt_options to True  , and for CreateExecutionProviderFactory(device_id) , set it to false.